core/iwasm/common/wasm_shared_memory.c

```diff
@@ -113,14 +121,14 @@ notify_stale_threads_on_exception(WASMModuleInstanceCommon *module_inst)
     uint32 i = 0, total_elem_count = 0;
     uint64 total_elem_count_size = 0;
 
-    os_mutex_lock(&shared_memory_list_lock);
+    os_mutex_lock(&wait_map_lock);
```

Comment:
bh_hash_map_traverse is already locked:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/shared/utils/bh_hashmap.c#L317